packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -39,15 +40,49 @@ function HeaderToolbar() {
 		showIconLabels,
 		isListViewOpen,
 		listViewShortcut,
+		useKeyboardFocusShortcut,
 	} = useSelect( ( select ) => {
-		const { hasInserterItems, getBlockRootClientId, getBlockSelectionEnd } =
-			select( blockEditorStore );
+		const {
+			hasInserterItems,
+			getBlock,
+			getBlockRootClientId,
+			getBlockSelectionEnd,
+			getSelectedBlockClientId,
+			getFirstMultiSelectedBlockClientId,
+			getSettings,
+			__unstableGetEditorMode,
+		} = select( blockEditorStore );
 		const { getEditorSettings } = select( editorStore );
 		const { getEditorMode, isFeatureActive, isListViewOpened } =
 			select( editPostStore );
 		const { getShortcutRepresentation } = select( keyboardShortcutsStore );
+		const editorMode = __unstableGetEditorMode();
+		const isDistractionFree = getSettings().isDistractionFree;
+		const hasFixedToolbar = getSettings().hasFixedToolbar;
+
+		let isBlockWithToolbarSelected = false;
+		// Check if we have an empty block selected
+		// If the first block in a multi selection is empty, there isn't a toolbar to show
+		const selectedBlockId =
+			getSelectedBlockClientId() || getFirstMultiSelectedBlockClientId();
+		if ( selectedBlockId ) {
+			const { name, attributes = {} } = getBlock( selectedBlockId ) || {};
+			isBlockWithToolbarSelected =
+				name && ! isUnmodifiedDefaultBlock( { name, attributes } );
+		}
```

Comment:
I'm not sure if skipping the name check will break things, but it would be nice if this could be simplified.

```suggestion
		// Check if we have an empty block selected
		// If the first block in a multi selection is empty, there isn't a toolbar to show
		const selectedBlockId = getSelectedBlockClientId() || getFirstMultiSelectedBlockClientId()
		const isBlockWithToolbarSelected = selectedBlockId
			? ! isUnmodifiedDefaultBlock( getBlock( selectedBlockId ) )
			: false
```

Reply:
I agree -- that's simpler and it works!